Tests fail with Python 3.5

In a test rebuild on Ubuntu, some tests failed with Python 3.5:

```
======================================================================
FAIL: Doctest: imported_tests.rst
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python3.5/doctest.py", line 2189, in runTest
    raise self.failureException(self.format_failure(new.getvalue()))
AssertionError: Failed doctest test for imported_tests.rst
  File "/«PKGBUILDDIR»/build/tests/functional_tests/doc_tests/test_issue145/imported_tests.rst", line 0

----------------------------------------------------------------------
File "/«PKGBUILDDIR»/build/tests/functional_tests/doc_tests/test_issue145/imported_tests.rst", line 43, in imported_tests.rst
Failed example:
    run(argv=argv) # doctest: +REPORT_NDIFF
Differences (ndiff with -expected +actual):
      package1 setup
      test (package1.test_module.TestCase) ... ok
      package1.test_module.TestClass.test_class ... ok
      package1.test_module.test_function ... ok
      package2c setup
    - test (package2c.test_module.TestCase) ... ok
    ?                             ^^  ^  ^
    + test (package2c.test_module.transplant_class.<locals>.C) ... ok
    ?                             ^^^^ ++++ ^^^  ^^^^^^^^^^^^
      package2c.test_module.TestClass.test_class ... ok
      package2f setup
      package2f.test_module.test_function ... ok
      <BLANKLINE>
      ----------------------------------------------------------------------
      Ran 6 tests in ...s
      <BLANKLINE>
      OK
----------------------------------------------------------------------
File "/«PKGBUILDDIR»/build/tests/functional_tests/doc_tests/test_issue145/imported_tests.rst", line 72, in imported_tests.rst
Failed example:
    run(argv=argv) # doctest: +REPORT_NDIFF
Differences (ndiff with -expected +actual):
      package2c setup
    - test (package2c.test_module.TestCase) ... ok
    ?                             ^^  ^  ^
    + test (package2c.test_module.transplant_class.<locals>.C) ... ok
    ?                             ^^^^ ++++ ^^^  ^^^^^^^^^^^^
      package2c.test_module.TestClass.test_class ... ok
      <BLANKLINE>
      ----------------------------------------------------------------------
      Ran 2 tests in ...s
      <BLANKLINE>
      OK
----------------------------------------------------------------------
File "/«PKGBUILDDIR»/build/tests/functional_tests/doc_tests/test_issue145/imported_tests.rst", line 99, in imported_tests.rst
Failed example:
    run(argv=argv) # doctest: +REPORT_NDIFF
Differences (ndiff with -expected +actual):
      package2c setup
    - test (package2c.test_module.TestCase) ... ok
    ?                             ^^  ^  ^
    + test (package2c.test_module.transplant_class.<locals>.C) ... ok
    ?                             ^^^^ ++++ ^^^  ^^^^^^^^^^^^
      <BLANKLINE>
      ----------------------------------------------------------------------
      Ran 1 test in ...s
      <BLANKLINE>
      OK


======================================================================
FAIL: runTest (test_load_tests_from_test_case.TestLoadTestsFromTestCaseHook)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«PKGBUILDDIR»/build/tests/functional_tests/test_load_tests_from_test_case.py", line 52, in runTest
    self.assertEqual(line.strip(), expect.pop(0))
AssertionError: 'test[34 chars]case.NoFixturePlug.loadTestsFromTestCase.<loca[18 chars]RROR' != 'test[34 chars]case.Derived) ... ERROR'
- test_value (test_load_tests_from_test_case.NoFixturePlug.loadTestsFromTestCase.<locals>.Derived) ... ERROR
+ test_value (test_load_tests_from_test_case.Derived) ... ERROR

-------------------- >> begin captured stdout << ---------------------
options
configure
Called!
test_value (test_load_tests_from_test_case.NoFixturePlug.loadTestsFromTestCase.<locals>.Derived) ... ERROR
test_value (tests.Tests) ... ok

======================================================================
ERROR: test_value (test_load_tests_from_test_case.NoFixturePlug.loadTestsFromTestCase.<locals>.Derived)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«PKGBUILDDIR»/build/tests/functional_tests/support/ltftc/tests.py", line 9, in test_value
    self.assertEqual(self.value, 1)
AttributeError: 'Derived' object has no attribute 'value'

----------------------------------------------------------------------
Ran 2 tests in 0.016s

FAILED (errors=1)


--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: test_addError (test_xunit.TestXMLOutputWithXML)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«PKGBUILDDIR»/build/tests/unit_tests/test_xunit.py", line 193, in test_addError
    eq_(tc.attrib['classname'], "test_xunit.TC")
AssertionError: 'test_xunit.mktest.<locals>.TC' != 'test_xunit.TC'
-------------------- >> begin captured stdout << ---------------------
b'<?xml version="1.0" encoding="UTF-8"?><testsuite name="nosetests" tests="1" errors="1" failures="0" skip="0"><testcase classname="test_xunit.mktest.&lt;locals&gt;.TC" name="runTest" time="0.000"><error type="builtins.RuntimeError" message="some error happened"><![CDATA[Traceback (most recent call last):\n  File "/«PKGBUILDDIR»/build/tests/unit_tests/test_xunit.py", line 175, in test_addError\n    raise RuntimeError("some error happened")\nRuntimeError: some error happened\n]]></error></testcase></testsuite>'

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: test_addFailure (test_xunit.TestXMLOutputWithXML)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«PKGBUILDDIR»/build/tests/unit_tests/test_xunit.py", line 126, in test_addFailure
    eq_(tc.attrib['classname'], "test_xunit.TC")
AssertionError: 'test_xunit.mktest.<locals>.TC' != 'test_xunit.TC'
-------------------- >> begin captured stdout << ---------------------
b'<?xml version="1.0" encoding="UTF-8"?><testsuite name="nosetests" tests="1" errors="0" failures="1" skip="0"><testcase classname="test_xunit.mktest.&lt;locals&gt;.TC" name="runTest" time="0.000"><failure type="builtins.AssertionError" message="one is not \'equal\' to two"><![CDATA[Traceback (most recent call last):\n  File "/«PKGBUILDDIR»/build/tests/unit_tests/test_xunit.py", line 108, in test_addFailure\n    raise AssertionError("one is not \'equal\' to two")\nAssertionError: one is not \'equal\' to two\n]]></failure></testcase></testsuite>'

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: test_addSuccess (test_xunit.TestXMLOutputWithXML)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/«PKGBUILDDIR»/build/tests/unit_tests/test_xunit.py", line 282, in test_addSuccess
    eq_(tc.attrib['classname'], "test_xunit.TC")
AssertionError: 'test_xunit.mktest.<locals>.TC' != 'test_xunit.TC'
-------------------- >> begin captured stdout << ---------------------
b'<?xml version="1.0" encoding="UTF-8"?><testsuite name="nosetests" tests="1" errors="0" failures="0" skip="0"><testcase classname="test_xunit.mktest.&lt;locals&gt;.TC" name="runTest" time="0.000"></testcase></testsuite>'

--------------------- >> end captured stdout << ----------------------
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,6 +5,7 @@ python:
   - 3.2
   - 3.3
   - 3.4
+  - 3.5
   - pypy
 install:
   - pip uninstall -y nose

--- a/functional_tests/test_load_tests_from_test_case.py
+++ b/functional_tests/test_load_tests_from_test_case.py
@@ -29,6 +29,7 @@ class NoFixturePlug(Plugin):
                 pass
             def tearDown(self):
                 pass
+        Derived.__qualname__ = Derived.__name__
         # must use nose loader here because the default loader in 2.3
         # won't load tests from base classes
         l = loader.TestLoader()

--- a/nose/util.py
+++ b/nose/util.py
@@ -643,6 +643,7 @@ def transplant_class(cls, module):
         pass
     C.__module__ = module
     C.__name__ = cls.__name__
+    C.__qualname__ = cls.__name__
     return C
 
 

--- a/unit_tests/test_xunit.py
+++ b/unit_tests/test_xunit.py
@@ -16,6 +16,7 @@ def mktest():
     class TC(unittest.TestCase):
         def runTest(self):
             pass
+    TC.__qualname__ = TC.__name__
     test = TC()
     return test
 

